### PR TITLE
Add common libarchive headers + cleanup a few misc things

### PR DESCRIPTION
**Pull request by L1ghtmann**
What does this implement/fix? Explain your changes.
---------------------------------------------------
1) Adds archive.h and archive_entry.h from the [libarchive](https://github.com/libarchive/libarchive/tree/master/libarchive) project 

2) Adjusts the formatting for Spark's headers so they're consistent 

3) Removes an old gitignore from rocketbootstrap/

4) Fixes indentation for a couple lines in CaptainHook.h

5) Removes the "headers" distinction from a few lines in the license file (I feel like it's likely implied from the repo)

6) Fixes a couple typos in the two bootstrap.h files

Checklist
---------
- [x] New code follows the [code rules](https://github.com/theos/headers/blob/master/README.md#code-rules)
- [ ] I've added modulemaps for any new libraries (e.g. see [libactivator/module.modulemap](https://github.com/theos/headers/blob/f3e596d896bae8f07c43cfb00ef55bf6224b4cdc/libactivator/module.modulemap)): it should be possible to `@import MyLibrary;` in ObjC, or `import MyLibrary` in Swift.
- [x] My contribution is code written by myself from reverse-engineered headers, licensed into the Public Domain as per [LICENSE.md](LICENSE.md); or, code written by myself / taken from an existing project, released under an OSI-approved license, and I've added relevant licensing credit to [LICENSE.md](LICENSE.md)

Does this close any currently open issues?
------------------------------------------
Nope

Any relevant logs, error output, etc?
-------------------------------------
N/A

Any other comments?
-------------------
These are the only two headers I've used personally, but there are a handful of others in the project that are similarly licensed that I can add if desired. 

No edits were made to the headers added.

Where has this been tested?
---------------------------
**Operating System:** …

Linux (WSL)

**Platform:** …

**Target Platform:** …

**Toolchain Version:** …

**SDK Version:** …

### CODE REVIEW

**Review by kirb — APPROVED** [review]

File: SparkAppItem.h [hunk 1]
```diff
@@ -1,6 +1,6 @@
 @interface SparkAppItem : NSObject
-    @property (nonatomic, retain) NSString* bundleIdentifier;
-    @property (nonatomic, retain) NSString* displayName;
+@property (nonatomic, retain) NSString* bundleIdentifier;
+@property (nonatomic, retain) NSString* displayName;
 
-    -(UIImage*)icon;
-@end
\ No newline at end of file
+-(UIImage*)icon;
```
Comment (PoomSmart) [inline SparkAppItem.h:5]:
```suggestion
- (UIImage*)icon;
```

File: SparkAppList.h [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-    id handler;
+	id handler;
 }
-    -(void)getAppList:(void(^)(NSArray*))handler;
+-(void)getAppList:(void(^)(NSArray*))handler;
 
-    +(NSArray*)getAppListForIdentifier:(NSString*)identifier andKey:(NSString*)key;
-    +(BOOL)doesIdentifier:(NSString*)identifier andKey:(NSString*)key containBundleIdentifier:(NSString*)bundleIdentifier;
-    +(void)setAppList:(NSArray*)appList forIdentifier:(NSString*)identifier andKey:(NSString*)key;
++(NSArray*)getAppListForIdentifier:(NSString*)identifier andKey:(NSString*)key;
```
Comment (PoomSmart) [inline SparkAppList.h:7]:
```suggestion
+ (NSArray*)getAppListForIdentifier:(NSString*)identifier andKey:(NSString*)key;
```
and friends below

File: SparkAppListTableViewController.h [hunk 3]
```diff
@@ -11,5 +11,4 @@
 
 -(id)initWithIdentifier:(NSString*)identifier andKey:(NSString*)key;
```
Comment (PoomSmart) [inline SparkAppListTableViewController.h:12]:
```suggestion
- (id)initWithIdentifier:(NSString*)identifier andKey:(NSString*)key;
```
and a friend below

File: SparkAppItem.h (hunk 1 above)
Comment (L1ghtmann) [reply SparkAppItem.h]: Done and also edited the others to match.

**Review by PoomSmart — APPROVED** [review]

**Pull request merged**